src/workspaces/features/RunPipelineDialog/RunPipelineDialog.tsx

```diff
@@ -33,12 +33,14 @@ type RunPipelineDialogProps = {
 const RunPipelineDialog = (props: RunPipelineDialogProps) => {
   const router = useRouter();
   const { open, onClose, pipeline } = props;
+  const [showVersionPicker, setShowVersionPicker] = useState(false);
   const clearCache = useCacheKey(["pipelines", pipeline.code]);
 
   const form = useForm<{ version: PipelineVersion; [key: string]: any }>({
     async onSubmit(values) {
       const { version, ...params } = values;
-      const run = await runPipeline(pipeline.id, params, version?.number);
+      const versionNumber = showVersionPicker ? version.number : undefined;
```

Comment:
This implementation does not cover the case where the user wants to run again a run since the showVersionPicker will not be true.

Maybe it's easier to populate the version if a run is provided and nothing otherwise. it removes this condition as well